src/main/java/com/google/api/codegen/config/GrpcGapicRetryMapping.java

```diff
@@ -0,0 +1,144 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.config;
+
+import com.google.api.codegen.RetryCodesDefinitionProto;
+import com.google.api.codegen.RetryParamsDefinitionProto;
+import com.google.api.codegen.grpc.MethodConfig.Name;
+import com.google.api.codegen.grpc.MethodConfig.RetryOrHedgingPolicyCase;
+import com.google.api.codegen.grpc.MethodConfig.RetryPolicy;
+import com.google.api.codegen.grpc.ServiceConfig;
+import com.google.api.tools.framework.model.Interface;
+import com.google.auto.value.AutoValue;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableMap;
+import com.google.protobuf.util.Durations;
+import java.util.HashMap;
+import java.util.Map;
+
+@AutoValue
+public abstract class GrpcGapicRetryMapping {
+
+  public static GrpcGapicRetryMapping create(
+      ServiceConfig config, ImmutableMap<String, Interface> protoInterfaces) {
+    Map<String, String> methodCodesMap = new HashMap<>();
+    Map<String, String> methodParamsMap = new HashMap<>();
+    Map<String, RetryCodesDefinitionProto> codesDefMap = new HashMap<>();
+    Map<String, RetryParamsDefinitionProto> paramsDefMap = new HashMap<>();
+
+    // build retry-to-interface mapping from gRPC ServiceConfig
+    int retryNdx = 1;
+    for (com.google.api.codegen.grpc.MethodConfig mc : config.getMethodConfigList()) {
```

Comment:
nit (very minor): Java generally prefers "real words" rather than abbreviations for variable names. e.g., `config` here would be preferred over `mc`. If you have time, changing some of these cases in this method would be really great. They are still correct as they are now for sure.

Reply:
Yes, definitely. Sorry that's the gopher in me using short variable names. I will change them everywhere :smile_cat:

Reply:
Done.